node/chain_submodule.go

```diff
@@ -0,0 +1,32 @@
+package node
+
+import (
+	"context"
+
+	"github.com/filecoin-project/go-filecoin/chain"
+	"github.com/filecoin-project/go-filecoin/consensus"
+	"github.com/filecoin-project/go-filecoin/net"
+	"github.com/filecoin-project/go-filecoin/net/pubsub"
+	"github.com/filecoin-project/go-filecoin/util/moresync"
+)
+
+// ChainSubmodule enhances the `Node` with chain capabilities.
+type ChainSubmodule struct {
+	BlockSub     pubsub.Subscription
+	Consensus    consensus.Protocol
+	ChainReader  nodeChainReader
+	MessageStore *chain.MessageStore
+	Syncer       nodeChainSyncer
```

Comment:
Move `nodeChainReader` and `nodeChainReader` to this file.

Reply:
If it's all the same with you, I rather move code closer to where it belongs on a followup PR.